File: src/odoc_240109/Solution.py
def solution(elements):
    sumList = []
    slice = []
    index = 0
    for i in range(len(elements)):
        for count in range(len(elements)):
            if(count + index >= len(elements)):
                index -= len(elements)
            slice.append(elements[count+index])
            sumList.append(sum(slice))
            if(len(slice)==len(elements)):
                slice.clear()
        index += 1

    sumList = list(set(sumList))
    answer = len(sumList)
    return answer

result = solution([7,9,1,1,4])
print(result)

File: src/odoc_240109/Solution.java
package odoc_240109;

import java.util.*;

public class Solution {
	public int solution(int[] elements) {
		
		List<Integer> sumList = new ArrayList<Integer>();
		List<Integer> slice = new ArrayList<Integer>();
		int index = 0;
		for(int i=0; i<elements.length; i++) {
			for(int j=0; j<elements.length; j++) {
				if(j + index >= elements.length) slice.add(elements[j+index-elements.length]);
				else slice.add(elements[j+index]);
				int sum = 0;
				for(int sliceSum:slice) {
					sum += sliceSum;
				}
				sumList.add(sum);
				if(slice.size()==elements.length) slice.clear();
			}
	        
			index += 1;
			
		}
			            
		sumList = new ArrayList<Integer>(new HashSet<Integer>(sumList));
		int answer = sumList.size();
		return answer;
    }

}
